apps/fyler/src/fyler_category_manager.erl

```diff
@@ -0,0 +1,694 @@
+      (Id, {Map, Nodes}) ->
+        case re:run(aws_cli:instance(Id), Re, [{capture, [ip], list}]) of
+          {match, [Ip]} ->
+            Node = ?NODE(Category, Ip),
+            {maps:put(Node, Id, Map), [Node | Nodes]};
+          nomatch ->
+            {Map, Nodes}
+        end
+    end,
+  {NodeToId, AllNodes} = lists:foldl(IdToNode, {#{}, []}, InstanceList),
+  PassiveNodes = lists:subtract(AllNodes, Pools),
+  NewState = State#state{pools = Pools, passive_pools = PassiveNodes, priorities = Priorities, node_to_id = NodeToId},
+  {next_state, NextState, NewState}.
+
+zero_(start, State) ->
+  {next_state, zero, State};
+
+zero_({run, #task{priority = Priority} = Task}, #state{tasks = Tasks, priorities = Priorities} = State) ->
+  NumPriority = maps:get(Priority, Priorities, 1),
+  update_ets(Task),
+  {next_state, tasks, State#state{tasks = fyler_queue:in(Task, NumPriority, Tasks)}};
+
+zero_({task_rejected, Ref, Node}, #state{category = Category, retasks = ReTasks, pools = Pools} = State) ->
+  case status(Node, Pools) of
+    {pending, Ref, Id} ->
+      NewPools = enable(Node, Pools),
+      case task(Category, Id) of
+        undefined ->
+          {next_state, zero, State#state{pools = NewPools}};
+        Task ->
+          {NextState, NewState} = choose_pool(State#state{retasks = fyler_queue:in(Task, ReTasks), pools = NewPools}),
+          {next_state, NextState, NewState}
+      end;
+    _ ->
+      {next_state, zero, State}
+  end;
+
+zero_({pool_connected, Node}, #state{pools = Pools, passive_pools = PassivePools} = State) ->
+  {next_state, pools, State#state{pools = connect(Node, Pools), passive_pools = lists:delete(Node, PassivePools)}};
+
+zero_({pool_enabled, Node}, #state{pools = Pools} = State) ->
+  NewPools = enable(Node, Pools),
+  {next_state, pools, State#state{pools = NewPools}};
+
+zero_({task_accepted, Ref, Node}, #state{pools = Pools} = State) ->
+  case status(Node, Pools) of
+    {pending, Ref, _Id} ->
+      NewPools = enable(Node, Pools),
+      {next_state, pools, State#state{pools = NewPools}};
+    _ ->
+      {next_state, zero, State}
+  end;
+
+zero_({pool_disabled, Node}, #state{category = Category, retasks = ReTasks, pools = Pools} = State) ->
+  NewPools = disable(Node, Pools),
+  case status(Node, Pools) of
+    {pending, _Ref, Id} ->
+      case task(Category, Id) of
+        undefined ->
+          {next_state, zero, State#state{pools = NewPools}};
+        Task ->
+          rpc:cast(Node, fyler_pool, cancel_task, [Id]),
+          {next_state, tasks, State#state{retasks = fyler_queue:in(Task, ReTasks), pools = NewPools}}
+      end;
+    _ ->
+      {next_state, zero, State#state{pools = NewPools}}
+  end;
+
+zero_({pool_down, Node}, #state{category = Category, retasks = ReTasks, pools = Pools,
+  passive_pools = PassivePools} = State) ->
+
+  NewReTasks = restart_tasks(Category, ReTasks, Node),
+  NewState = State#state{pools = delete(Node, Pools), passive_pools = [Node|PassivePools]},
```

Comment:
Line 235 is too long:   NewState = State#state{pools = delete(Node, Pools), passive_pools = [Node|PassivePools]},.